apps/ivs-functions/src/handlers/recorderEndedJsonHandler.ts

```
@@ -0,0 +1,93 @@
+    }
+
+    const endRecordingUrl = `${videoIvsBasePath}/${key}`;
+
+    const s3Response = await fetch(endRecordingUrl);
+
+    if (!s3Response.ok) {
+      // eslint-disable-next-line functional/no-throw-statements
+      throw new Error('Failed to fetch recording-ended.json from S3');
+    }
+    const recordingEndedFile: RecordingEndedFile = await s3Response.json();
+
+    console.log(
+      'recordingEndedFile',
+      JSON.stringify(recordingEndedFile, null, 2)
+    );
+
+    const startStreamingTime = new Date(
+      recordingEndedFile.recording_started_at
+    ).getTime();
+    const hls = recordingEndedFile.media.hls;
+
+    const newPlayerSrc = [
+      videoIvsBasePath,
+      ...key.split('/').slice(0, -2), // get to base path of VOD removing last two segments
+      hls.path,
+      hls.playlist,
+    ].join('/');
+
+    const strapiWebinarsUrl = `api/webinars?filters[startDatetime][$gte]=${new Date(
+      startStreamingTime - START_STREAMING_TRASHOLD_MS
+    ).toISOString()}&filters[startDatetime][$lte]=${new Date(
+      startStreamingTime + START_STREAMING_TRASHOLD_MS
+    ).toISOString()}`;
+
+    const strapiWebinarsResponse: StrapiWebinars =
+      await fetchFromStrapiResponse(strapiWebinarsUrl);
+
+    if (strapiWebinarsResponse.meta.pagination.total !== 1) {
+      // eslint-disable-next-line functional/no-throw-statements
+      throw new Error(
+        `Expected to find exactly one webinar for startDatetime around ${recordingEndedFile.recording_started_at}, but found ${strapiWebinarsResponse.meta.pagination.total}`
+      );
+    }
+    const webinar = strapiWebinarsResponse.data[0];
+
+    await fetchFromStrapiResponse(`api/webinars/${webinar.id}`, {
+      method: 'PUT',
+      body: {
+        data: {
+          playerSrc: newPlayerSrc,
+        },
+      },
+    });
+
+    return event;
+  } catch (error) {
+    console.error('[EndStramingEvent] Error:', error);
```

Is this the desired behaviour? if anything bad happens it logs the error and just returns the event, suppressing the error and just logging it